packages/ckeditor5-dev-release-tools/tests/tasks/preparerepository.js

```diff
@@ -0,0 +1,321 @@
+				}
+
+				expect( stubs.fs.writeJson.callCount ).to.equal( 0 );
+				expect( stubs.fs.copy.callCount ).to.equal( 0 );
+			} );
+		} );
+
+		describe( 'monorepository packages processing', () => {
+			it( 'should copy files of all packages', async () => {
+				options.packagesDirectory = 'packages';
+
+				stubs.fs.lstat.withArgs( 'current/working/dir/packages/ckeditor5-core' ).resolves( stubs.lstat.isDir );
+				stubs.fs.lstat.withArgs( 'current/working/dir/packages/ckeditor5-utils' ).resolves( stubs.lstat.isDir );
+				stubs.fs.existsSync.withArgs( 'current/working/dir/packages/ckeditor5-core/package.json' ).resolves( true );
+				stubs.fs.existsSync.withArgs( 'current/working/dir/packages/ckeditor5-utils/package.json' ).resolves( true );
+
+				await prepareRepository( options );
+
+				expect( stubs.fs.copy.callCount ).to.equal( 2 );
+
+				expect( stubs.fs.copy.getCall( 0 ).args.length ).to.equal( 2 );
+				expect( stubs.fs.copy.getCall( 0 ).args[ 0 ] ).to.equal( 'current/working/dir/packages/ckeditor5-core' );
+				expect( stubs.fs.copy.getCall( 0 ).args[ 1 ] ).to.equal( 'current/working/dir/release/ckeditor5-core' );
+
+				expect( stubs.fs.copy.getCall( 1 ).args.length ).to.equal( 2 );
+				expect( stubs.fs.copy.getCall( 1 ).args[ 0 ] ).to.equal( 'current/working/dir/packages/ckeditor5-utils' );
+				expect( stubs.fs.copy.getCall( 1 ).args[ 1 ] ).to.equal( 'current/working/dir/release/ckeditor5-utils' );
+			} );
+
+			it( 'should not copy non-directories', async () => {
+				options.packagesDirectory = 'packages';
+
+				stubs.fs.readdir.withArgs( 'current/working/dir/packages' ).resolves( [ ...packages, 'textFile.txt' ] );
+
+				stubs.fs.lstat.withArgs( 'current/working/dir/packages/ckeditor5-core' ).resolves( stubs.lstat.isDir );
+				stubs.fs.lstat.withArgs( 'current/working/dir/packages/ckeditor5-utils' ).resolves( stubs.lstat.isDir );
+				stubs.fs.lstat.withArgs( 'current/working/dir/packages/textFile.txt' ).resolves( stubs.lstat.isNotDir );
+				stubs.fs.existsSync.withArgs( 'current/working/dir/packages/ckeditor5-core/package.json' ).resolves( true );
+				stubs.fs.existsSync.withArgs( 'current/working/dir/packages/ckeditor5-utils/package.json' ).resolves( true );
+
+				await prepareRepository( options );
+
+				expect( stubs.fs.copy.callCount ).to.equal( 2 );
+
+				expect( stubs.fs.copy.getCall( 0 ).args.length ).to.equal( 2 );
+				expect( stubs.fs.copy.getCall( 0 ).args[ 0 ] ).to.equal( 'current/working/dir/packages/ckeditor5-core' );
+				expect( stubs.fs.copy.getCall( 0 ).args[ 1 ] ).to.equal( 'current/working/dir/release/ckeditor5-core' );
+
+				expect( stubs.fs.copy.getCall( 1 ).args.length ).to.equal( 2 );
+				expect( stubs.fs.copy.getCall( 1 ).args[ 0 ] ).to.equal( 'current/working/dir/packages/ckeditor5-utils' );
+				expect( stubs.fs.copy.getCall( 1 ).args[ 1 ] ).to.equal( 'current/working/dir/release/ckeditor5-utils' );
```

Comment:
Instead of checking what has been copied, I would check if the `'current/working/dir/packages/textFile.txt'` file was not touched.